### PR TITLE
Added error handler for PHPParser errors

### PR DESCRIPTION
**Pull request by mikhainin**
Currently, if parser is failed it quite hard to get what exactly goes wrong.
Just added filename to error message to make it easier to locate the problem

### CODE REVIEW

File: src/Badoo/SoftMocks.php [hunk 1]
```diff
@@ -387,6 +387,26 @@ protected function pEncapsList(array $encapsList, $quote)
     }
 }
 
+class SoftMocksParseError extends \PhpParser\Error
+{
+    public function __construct(string $file_path, \PhpParser\Error $Error)
+    {
+        parent::__construct("File: {$orig_file}, message: {$error->getMessage()}", $Error->getAttributes());
```
Comment (Sannis) [inline src/Badoo/SoftMocks.php]:
```suggestion
        parent::__construct("File: {$file_path}, message: {$error->getMessage()}", $Error->getAttributes());
```

**Comment by mougrim** [comment]
@negram, what about write test with check that error handling is worked?

**Comment by bas-lua** [comment]
sure thing. will do

**Comment by mikhainin** [comment]
sure thing. will do

**Pull request merged**